src/compute/src/logging/compute.rs

```diff
@@ -297,22 +300,27 @@ pub fn construct<A: Allocate>(
                         .collect::<Vec<_>>(),
                     variant.desc().arity(),
                 );
-                let trace = collection
-                    .map({
-                        let mut row_buf = Row::default();
-                        let mut datums = DatumVec::new();
-                        move |row| {
-                            let datums = datums.borrow_with(&row);
-                            row_buf.packer().extend(key.iter().map(|k| datums[*k]));
-                            let row_key = row_buf.clone();
-                            row_buf.packer().extend(value.iter().map(|k| datums[*k]));
-                            let row_val = row_buf.clone();
-                            (row_key, row_val)
-                        }
-                    })
+                let rows = collection.map({
+                    let mut row_buf = Row::default();
+                    let mut datums = DatumVec::new();
+                    move |row| {
+                        let datums = datums.borrow_with(&row);
+                        row_buf.packer().extend(key.iter().map(|k| datums[*k]));
+                        let row_key = row_buf.clone();
+                        row_buf.packer().extend(value.iter().map(|k| datums[*k]));
+                        let row_val = row_buf.clone();
+                        (row_key, row_val)
+                    }
+                });
+
+                let trace = rows
                     .arrange_named::<RowSpine<_, _, _, _>>(&format!("ArrangeByKey {:?}", variant))
                     .trace;
-                result.insert(variant, (trace, Rc::clone(&token)));
+                result.insert(variant.clone(), (trace, Rc::clone(&token)));
+            }
+
+            if let Some(target) = config.sink_logs.get(&variant) {
+                persist_sink(target.0, &target.1, compute_state, &collection);
```

Comment:
You could the target parts readable names, to make it more clear what they are. E.g.:
```
if let Some((id, storage_metadata)) = config.sink_logs.get(&variant) {
    persist_sink(id, storage_metadata, compute_state, &collection);
}
```